Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/ImportOneProcessController.java

```diff
@@ -154,6 +173,76 @@ public void importProcess(final String domain, final String owner)
     }
   }
 
+  private void importLinkedSubProcesses(final ProcessSummaryType importedModel, final BPMNDiagram bpmnDiagram,
+                                  final String domain, final String owner, final String name, final int folderId)
+      throws Exception {
+    Map<SubProcess, ProcessSummaryType> subProcessMap = new HashMap<>();
+    Map<ProcessSummaryType, Map<BPMNNode, BPMNNode>> subProcessModelClonedNodesMap = new HashMap<>();
+    int count = 0;
+    for (SubProcess subProcess : bpmnDiagram.getSubProcesses()) {
+      BPMNDiagram subProcessDiagram = BPMNDiagramFactory.newBPMNDiagram("");
+      Map<BPMNNode, BPMNNode> nodeMap = subProcessDiagram.cloneSubProcessContents(subProcess);
+      String subProcessName = name + "_subprocess" + ++count;
+      ProcessSummaryType subProcessSummary = importSubProcess(subProcessDiagram, domain, owner, subProcessName, folderId);
+      subProcessMap.put(subProcess, subProcessSummary);
+      subProcessModelClonedNodesMap.put(subProcessSummary, nodeMap);
+    }
+
+    linkSubProcesses(importedModel, subProcessMap, subProcessModelClonedNodesMap);
+  }
+
+  private ProcessSummaryType importSubProcess(final BPMNDiagram bpmnDiagram, final String domain, final String owner,
+                                              final String name, final int folderId)
+      throws Exception {
+    String emptyModelXML = "<?xml version='1.0' encoding='UTF-8'?>"
+        + "<bpmn:definitions xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance' "
+        + "xmlns:bpmn='http://www.omg.org/spec/BPMN/20100524/MODEL' "
+        + "xmlns:bpmndi='http://www.omg.org/spec/BPMN/20100524/DI' "
+        + "xmlns:dc='http://www.omg.org/spec/DD/20100524/DC' "
+        + "targetNamespace='http://bpmn.io/schema/bpmn' " + "id='Definitions_1'>"
+        + "<bpmn:process id='Process_1' isExecutable='false'/>"
+        + "<bpmndi:BPMNDiagram id='BPMNDiagram_1'>"
+        + "<bpmndi:BPMNPlane id='BPMNPlane_1' bpmnElement='Process_1'/>"
+        + "</bpmndi:BPMNDiagram>"
+        + "</bpmn:definitions>";
+
+    String bpmnText = bpmnDiagram.getNodes().isEmpty() ? emptyModelXML : BpmnLayoutPlugin.addLayout(bpmnDiagram, "");
+    ImportProcessResultType importResult = mainC.getManagerService().importProcess(owner,
```

Comment:
As discussed, this PR needs a batch import operation because it imports multiple diagrams (each for one sub-process) and they are all related. Any error happens will leave the database in an unknown state with some garbage sub-processes already imported. Imagine, it would be a nightmare to clean up the database if it happens to a large digaram with many sub-processes.

Reply:
As discussed, the batch import will be handled as part of a further improvement, seperate from this PR. Marking this comment as resolved.